File: Frontend/sheikh-travel-control-center/src/app/core/navigation/tenant-type.ts
export enum TenantType {
  TravelAgency = 'travelAgency',
  FleetOperator = 'fleetOperator',
  CorporateCustomer = 'corporateCustomer',
  Driver = 'driver'
}

export function resolveTenantType(
  roles: string[],
  override?: TenantType
): TenantType {
  if (override) return override;
  if (roles.includes('Driver')) return TenantType.Driver;
  if (roles.includes('FleetManager')) return TenantType.FleetOperator;
  if (roles.includes('CorporateUser')) return TenantType.CorporateCustomer;
  return TenantType.TravelAgency;
}

export function tenantProductSubtitle(type: TenantType): string {
  switch (type) {
    case TenantType.FleetOperator:
      return 'FLEET CONTROL';
    case TenantType.CorporateCustomer:
      return 'CORPORATE PORTAL';
    case TenantType.Driver:
      return 'DRIVER APP';
    default:
      return 'CONTROL CENTER';
  }
}

File: Frontend/sheikh-go/lib/core/navigation/tenant_type.dart
enum TenantType {
  travelAgency,
  fleetOperator,
  corporateCustomer,
  driver,
}

extension TenantTypeX on TenantType {
  String get label => switch (this) {
        TenantType.travelAgency => 'Travel Agency',
        TenantType.fleetOperator => 'Fleet Operator',
        TenantType.corporateCustomer => 'Corporate Client',
        TenantType.driver => 'Driver',
      };

  String get productSubtitle => switch (this) {
        TenantType.travelAgency => 'CONTROL CENTER',
        TenantType.fleetOperator => 'FLEET CONTROL',
        TenantType.corporateCustomer => 'CORPORATE PORTAL',
        TenantType.driver => 'DRIVER APP',
      };

  String get accountTierLabel => switch (this) {
        TenantType.travelAgency => 'Premium Account',
        TenantType.fleetOperator => 'Fleet Enterprise',
        TenantType.corporateCustomer => 'Corporate Account',
        TenantType.driver => 'Driver Account',
      };
}

TenantType resolveTenantType({
  required List<String> roles,
  TenantType? override,
}) {
  if (override != null) return override;
  if (roles.contains('Driver')) return TenantType.driver;
  if (roles.contains('FleetManager')) return TenantType.fleetOperator;
  if (roles.contains('CorporateUser')) return TenantType.corporateCustomer;
  return TenantType.travelAgency;
}
